src/celeritas/field/FieldPropagator.hh

```diff
@@ -239,10 +257,35 @@ CELER_FUNCTION auto FieldPropagator<DriverT>::operator()(real_type step)
     normalize_direction(&dir);
     geo_.set_dir(dir);
 
+    if (CELER_UNLIKELY(result.distance == 0))
+    {
+        // We failed to move at all, which means we hit a boundary no matter
+        // what step length we took, which means we're stuck.
+        // Using the just-reapplied direction, hope that we're pointing deeper
+        // into the current volume and bump the particle.
+        axpy(this->bump_distance(), dir, &state_.pos);
+        geo_.move_internal(state_.pos);
+        result.distance = this->bump_distance();
+        result.boundary = false;
+    }
+
     CELER_ENSURE(result.boundary == geo_.is_on_boundary());
     CELER_ENSURE(result.distance > 0 && result.distance <= step);
```

Comment:
I'm hitting this assertion with a track starting on a boundary and reentrant with a step length smaller than the bump distance. Is it ok to relax the second expression to allow the distance to be larger than the step if it's equal to the bump?

Reply:
Just curious why a step length smaller than the bump distance (<= minimum_step of the field driver?) is propagated in a magnetic field.

Reply:
@sethrj yeah that makes more sense I think